csharp/test/Drivers/Apache/Spark/ThriftClientAsyncMock.cs

```diff
@@ -0,0 +1,135 @@
+
+            public Task<TGetOperationStatusResp> GetOperationStatus(TGetOperationStatusReq req, CancellationToken cancellationToken = default) => throw new NotImplementedException();
+
+            public Task<TGetPrimaryKeysResp> GetPrimaryKeys(TGetPrimaryKeysReq req, CancellationToken cancellationToken = default) => throw new NotImplementedException();
+
+            public Task<TGetQueryIdResp> GetQueryId(TGetQueryIdReq req, CancellationToken cancellationToken = default) => throw new NotImplementedException();
+
+            public Task<TGetResultSetMetadataResp> GetResultSetMetadata(TGetResultSetMetadataReq req, CancellationToken cancellationToken = default) => throw new NotImplementedException();
+
+            public Task<TGetSchemasResp> GetSchemas(TGetSchemasReq req, CancellationToken cancellationToken = default) => throw new NotImplementedException();
+
+            public Task<TGetTablesResp> GetTables(TGetTablesReq req, CancellationToken cancellationToken = default) => throw new NotImplementedException();
+
+            public Task<TGetTableTypesResp> GetTableTypes(TGetTableTypesReq req, CancellationToken cancellationToken = default)
+            {
+                TStatus status = new(TStatusCode.SUCCESS_STATUS);
+                StringArray.Builder stringBuilder = new();
+                stringBuilder.Append("TABLE");
+                stringBuilder.Append("VIEW");
+                TStringColumn stringColumn = new(stringBuilder.Build());
+                TTypeDesc stringTypeDesc = new()
+                {
+                    Types = [ new TTypeEntry() { PrimitiveEntry = new TPrimitiveTypeEntry(TTypeId.STRING_TYPE) } ]
+                };
+                TGetTableTypesResp resp = new(status)
+                {
+                    DirectResults = new()
+                    {
+                        ResultSet = new(status)
+                        {
+                            ResultSetMetadata = new(status)
+                            {
+                                Schema = new()
+                                {
+                                    Columns = [new TColumnDesc("table_type", stringTypeDesc, 0)],
+                                },
+                            },
+                            Results = new()
+                            {
+                                ColumnCount = 1,
+                                Columns = [new TColumn() { StringVal = stringColumn }],
+                            },
+                        },
+                    },
+                };
+
+                return Task.FromResult(resp);
+            }
+
+            public Task<TGetTypeInfoResp> GetTypeInfo(TGetTypeInfoReq req, CancellationToken cancellationToken = default) => throw new NotImplementedException();
+
+            public Task<TOpenSessionResp> OpenSession(TOpenSessionReq req, CancellationToken cancellationToken = default)
```

Comment:
This implementation should be fine to mock the `OpenSsession`. In fact, a recording of the values may not be desirable.